File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/security/AuthenticationSourceProvider.java
package net.nemerosa.ontrack.model.security;

/**
 * Defines the provider for an authentication source.
 *
 * @see net.nemerosa.ontrack.model.security.AuthenticationSource
 */
public interface AuthenticationSourceProvider {

    /**
     * Gets the source descriptor
     */
    AuthenticationSource getSource();

}

File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/security/AuthenticationSourceProvider.kt
package net.nemerosa.ontrack.model.security

/**
 * Defines the provider for an authentication source.
 *
 * @see net.nemerosa.ontrack.model.security.AuthenticationSource
 */
interface AuthenticationSourceProvider {

    /**
     * ID of this provider
     */
    val id: String

    /**
     * Gets the sources from this provider
     */
    val sources: List<AuthenticationSource>
}